lib/matplotlib/tests/test_animation.py

```diff
@@ -128,20 +134,72 @@ def animate(i):
         line.set_data(x, y)
         return line,
 
-    # Use NamedTemporaryFile: will be automatically deleted
-    F = tempfile.NamedTemporaryFile(suffix='.' + extension)
-    F.close()
-    anim = animation.FuncAnimation(fig, animate, init_func=init, frames=5)
+    with closed_tempfile(suffix='.' + extension) as fname:
+        anim = animation.FuncAnimation(fig, animate, init_func=init, frames=5)
+        anim.save(fname, fps=30, writer=writer, bitrate=500)
+
+
+@cleanup
+def check_save_animation_pep_519(writer, extension='mp4'):
+    class FakeFSPathClass(object):
+        def __init__(self, path):
+            self._path = path
+
+        def __fspath__(self):
+            return self._path
+
+    if not animation.writers.is_available(writer):
+        skip("writer '%s' not available on this system"
+                               % writer)
+    fig, ax = plt.subplots()
+    line, = ax.plot([], [])
+
+    ax.set_xlim(0, 10)
+    ax.set_ylim(-1, 1)
+
+    def init():
+        line.set_data([], [])
+        return line,
+
+    def animate(i):
+        x = np.linspace(0, 10, 100)
+        y = np.sin(x + i)
+        line.set_data(x, y)
+        return line,
+
+    with closed_tempfile(suffix='.' + extension) as fname:
+        anim = animation.FuncAnimation(fig, animate, init_func=init, frames=5)
+        anim.save(FakeFSPathClass(fname), fps=30, writer=writer, bitrate=500)
+
+
+@cleanup
+def check_save_animation_pathlib(writer, extension='mp4'):
```

Comment:
This test and the previous one looks like they could be refactoring to loop other Path and fspath compatible object to minimize the code duplication.